### PR TITLE
uint32_t support for AVX 

### PR DESCRIPTION
**Pull request by flori-uni**
Hi,

for my bachelors thesis I'm using this lib and so I started to implement some uint32_t support for AVX (AVX512 will follow)

I'm not sure if there is some interest on this implementation - so feel free to decline this PR :wink:

### CODE REVIEW

**Comment by kouchy** [comment]
Hi @flori-uni , 
yes it is very interesting, thx. The support of unsigned integers is missing in MIPP.
Perhaps I can create a specific branch named "uint" and accept your PR in this branch? What do you think?

**Comment by flori-uni** [comment]
Sure. I'll add further functionality the next days.

One short question by the way: Do you have something like a config file for an autoformatter so the code style stays consistent?

**Comment by kouchy** [comment]
No I don't use an autoformatter at this time. I simply try to follow some coding convention given here: https://github.com/aff3ct/aff3ct/blob/master/CONTRIBUTING.md#coding-conventions

**Comment by kouchy** [comment]
I created the `uint` branch and I enabled the tests on this branch you will see I you are breaking things on the GitLab pipeline here : https://gitlab.com/aff3ct/MIPP/pipelines.
Could you recreate your PR with the `uint` branch as the target branch please?

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ set (EXECUTABLE_OUTPUT_PATH bin/)
 #if(COMMAND cmake_policy)
 #    cmake_policy(SET CMP0003 NEW)
 #endif(COMMAND cmake_policy)
-
+add_definitions(-march=native)
```
Comment (kouchy) [inline tests/CMakeLists.txt]: I think you should avoid to put compiler specific options in the CMakeLists.txt file. For instance, the tests on Visual Studio compiler (MSVC) will not pass.

File: src/mipp_impl_AVX.hxx [hunk 2]
```diff
@@ -213,7 +240,11 @@
 		return _mm256_castsi256_ps(_mm256_set_epi32(vals[7], vals[6], vals[5], vals[4],
 		                                            vals[3], vals[2], vals[1], vals[0]));
 	}
-
+         template <>
+         inline reg set<uint32_t>(const uint32_t vals[nElReg<uint32_t>()]) {
+            return _mm256_castsi256_ps(_mm256_set_epi32(vals[7], vals[6], vals[5], vals[4],
```
Comment (kouchy) [inline src/mipp_impl_AVX.hxx]: As you use `_mm256_set_epi32` (expect `int32_t` type) you should add a casts to `uint32_t`.
`_mm256_set_epi32((uint32_t)vals[7], (uint32_t)vals[6], ...)`

File: src/mipp_impl_AVX.hxx [hunk 3]
```diff
@@ -295,7 +326,10 @@
 	inline reg set1<int32_t>(const int32_t val) {
 		return _mm256_castsi256_ps(_mm256_set1_epi32(val));
 	}
-
+         template <>
+         inline reg set1<uint32_t>(const uint32_t val) {
+            return _mm256_castsi256_ps(_mm256_set1_epi32(val));
```
Comment (kouchy) [inline src/mipp_impl_AVX.hxx]: You have to add a cast to `uint32_t` before to give val to `_mm256_set1_epi32`.

File: tests/src/logical_comparisons/cmpeq.cpp [hunk 4]
```diff
@@ -50,6 +50,7 @@ TEST_CASE("Compare equal - mipp::reg", "[mipp::cmpeq]")
 #endif
 #endif
 	SECTION("datatype = int32_t") { test_reg_cmpeq<int32_t>(); }
+        SECTION("datatype = uint32_t") { test_reg_cmpeq<uint32_t>(); }
```
Comment (kouchy) [inline tests/src/logical_comparisons/cmpeq.cpp]: Tabulations are used in MIPP, could you use tabs too :-)? This remark is valid for the whole PR.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (flori-uni) [reply tests/CMakeLists.txt]: Ok. Reason for this line was that by default GCC uses the lowest architecture (SSE) and I'm unsure whether to add unsigned functionality here...

File: tests/src/logical_comparisons/cmpeq.cpp (hunk 4 above)
Comment (flori-uni) [reply tests/src/logical_comparisons/cmpeq.cpp]: Oh i recognized that to late. My CLion had the wrong configuration. I'll fix it asap.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (kouchy) [reply tests/CMakeLists.txt]: If someone wants to compile in `SSE2` then MIPP will simply throw an exception saying that the called function using `uint32_t` is not available for this instruction set.

**Comment by flori-uni** [comment]
> I created the `uint` branch and I enabled the tests on this branch you will see if you are breaking things on the GitLab pipeline here : https://gitlab.com/aff3ct/MIPP/pipelines.
> Could you recreate your PR with the `uint` branch as the target branch please?

Will my PR trigger a CI build on GitLab? I'm not sure if that works?

**Comment by kouchy** [comment]
Yes when I will accept your PR, the CI will be automatically started on GitLab.com.

**Comment by flori-uni** [comment]
Ok. I will probably implement the 'basic' functionality and not every command.

**Comment by kouchy** [comment]
Yes of course, whatever you want ;-).

File: tests/CMakeLists.txt (hunk 1 above)
Comment (kouchy) [reply tests/CMakeLists.txt]: As expected, the CI fails with `-march=native` : https://gitlab.com/aff3ct/MIPP/-/jobs/380213921

Comment (flori-uni) [reply tests/CMakeLists.txt]: Yep... I've already changed my local configuration and planed to push it today.

**Pull request merged**